SPEC.md

```diff
@@ -0,0 +1,675 @@
+# Livepeer Technical Specification
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Design](#design)
+* [Users](#users)
+* [Data Types](#data-types)
+* [Faults](#faults)
+* [Contract API](#contract-api)
+* [Actions](#actions)
+* [Parameters](#parameters)
+
+## Introduction
+
+The Livepeer protocol defines the rules of participation for a cryptotoken powered decentralized live video transcoding marketplace. The initial version of the protocol is implemented via a system of smart contracts
+on the Ethereum blockchain.
+
+This document is a technical specification of the protocol. For a more general overview of the protocol, see the [whitepaper](https://github.com/livepeer/wiki/blob/master/WHITEPAPER.md).
+
+Given that the initial version of the protocol is implemented via a system of Ethereum smart contracts, the design of the protocol is necessarily influenced and restricted by the current capabilities of smart contracts
+and the Ethereum blockchain. Consequently, this specification assumes smart contracts as the implementation tool and will describe protocol actions in terms of smart contract function calls that
+trigger state transitions within the system.
+
+## Design
```

Comment:
I think just prior to the design section we should have a 1 or 2 paragraph overview of the progression of the protocol. I can try to write this. I think it will make it easier for someone to absorb the following technical concepts.